src/main/java/com/atlassian/bitbucket/jenkins/internal/status/BitbucketBuildStatusFactory.java

```diff
@@ -14,7 +14,7 @@
     private static final Collection<Result> successfulResults = Arrays.asList(Result.SUCCESS, Result.UNSTABLE);
 
     public static BitbucketBuildStatus fromBuild(Run<?, ?> build) {
-        String key = build.getId();
+        String key = build.getParent().getFullName() + " #" + build.getId();
```

Comment:
This should simple be build.getParent().getFullName(), essentially the same as what Bamboo does.

If the build number is included then subsequent runs add to the number of build results instead of replacing it. Imaging you run a build and it fails because of a flaky test, or maybe your build configuration is incorrect. When you re-run the build you expect the result to overwrite the old one, that way any merge checks now pass.

Reply:
I think this is actually what we want the name to be: https://docs.atlassian.com/bitbucket-server/rest/4.0.0/bitbucket-build-rest.html#idp59968
So swap those lines and this should be correct.

Reply:
As suggested by @benhumphreys, I have kept the `key` and `name` consistent with Bamboo. `key` will be the `Job` name and `name` will be `Job`'s display name.